src/theory/builtin/theory_builtin_rewriter.cpp

```diff
@@ -28,29 +30,36 @@ namespace cvc5::internal {
 namespace theory {
 namespace builtin {
 
-Node TheoryBuiltinRewriter::blastDistinct(TNode in) {
+Node TheoryBuiltinRewriter::blastDistinct(TNode in)
+{
   Assert(in.getKind() == kind::DISTINCT);
 
-  if(in.getNumChildren() == 2) {
+  NodeManager* nm = NodeManager::currentNM();
+
+  if (in[0].getType().isBitVector()
```

Comment:
This should not be hardcoded for bitvectors.

This could be done via `getType().getCardinality()`, however there are further caveats for this method to avoid performance issues.

Reply:
What are these caveats?